### PR TITLE
Replaced openssl-cert-tools by node-forge to validate pem certificates

### PR DESCRIPTION
**Pull request by cnadeau**
Fixes https://github.com/alexa-js/alexa-verifier/issues/11 by not using native openssl which seems to  introduce bugs in openssl-cert-tools as of 1.0.2k

### CODE REVIEW

**Comment by dblock** [comment]
I'm always worried about doing security code like cert validation. Sounds like a problem more than one person has, I wonder if there's a well tested package that does that and would remove the need for validate-cert.js? There're npm packages like https://www.npmjs.com/package/check-cert-expiry but I didn't look at it in depth.

**Comment by mreinstein** [comment]
> I'm always worried about doing security code like cert validation

I agree it's bad to write your own crypto routines, but we're not doing that. The current code is using openSSL, which is mostly reasonable except for the problems it's causing for windows/azure. other forms of validation are just a necessary part of application development, which is what we're doing. (e.g., parameter validation/sanitization/filtering.) I'd also argue that code should be written defensively, which by it's nature is implicitly security code. :)

> I wonder if there's a well tested package that does that and would 
> remove the need for validate-cert.js?

possibly. If there is such a library, I'm not opposed to using it provided it meets a sufficient level of quality. :) 

I think it's important to restate the goal of this module is to pass amazon's certification process, not to be a general purpose security pattern. So while I'm not opposed to doing more validation on certs, only to the extent that it doesn't interfere with that goal.

**Comment by cnadeau** [comment]
@dblock As mentioned above, it's to validate the CN of the cert is from Amazon and that it's not expired. But I agree usually we should not do any custom crypto logic.

@mreinstein sorry about all the updates, I'm not used to travis CI, but it's really awesome. And it now passes.

Any concern related to the usage of node-forge to convert the PEM to extract the public information and validate it?

**Comment by mreinstein** [comment]
@cnadeau 

> Any concern related to the usage of node-forge to convert the
> PEM to extract the public information and validate it?

on the surface it looks good but I'd love to get feedback from @Larochelle and @nazvam because they've got some overlap with their proposal in #1 What do you guys think?  

#meeting-of-the-minds

**Comment by cnadeau** [comment]
@mreinstein 
FYI: I'm the one taking over @Larochelle's work. He had to migrate from Azure web app to a docker container which solved the node environment issue.

**Comment by Larochelle** [comment]
+1 @cnadeau is your man. I'm not doing Alexa and Azure development these days.

File: validate-cert.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      return callback('certificate expiration check failed')
+    }
+
+    callback()
+  } catch (e) {
+    return callback(e);
+  }
+}
+
+module.exports = {
```
Comment (mreinstein) [inline validate-cert.js]: this module only exposes one function. I'd prefer to do 
```javascript
module.exports = function validate(pem_cert, callback) {
```

Comment (mreinstein) [inline validate-cert.js]: besides the small `module.exports` change, the rest of this PR LGTM. Nice work!

Comment (cnadeau) [inline validate-cert.js]: done, thank you for the lib!

**Pull request merged**